base/dict.jl

```diff
@@ -686,6 +686,18 @@ end
 
 filter!(f, d::Dict) = filter_in_one_pass!(f, d)
 
+function map!(f, iter::ValueIterator{<:Dict})
+    dict = iter.dict
+    vals = dict.vals
+    # @inbounds is here so the it gets propigated to isslotfiled
+    @inbounds for i = dict.idxfloor:lastindex(vals)
+        if isslotfilled(dict, i)
+            vals[i] = f(vals[i])
+        end
+    end
+    return vals
```

Comment:
I think we should return `iter` here? `vals` will contain a lot of uninitialized values, and this would adhere to the convention that `map!` returns the object that was mapped over.

Reply:
I have updated it to return iter